manifests/profile/networking/private.pp

```diff
@@ -19,7 +19,7 @@
 ) {
 
   if !$interface and $facts['os']['release']['major'] == '9' and $facts['is_virtual']
-    and $facts['networking']['interfaces'].has_key('ens4') {
+    and has_key($facts['networking']['interfaces'], 'ens4') {
```

Comment:
Still fails, message is:

```
  1) nebula::profile::networking::private on debian-9-x86_64 with no interface should not contain File[/etc/network/interfaces.d/private]
     Failure/Error: raise Puppet::ParseError, "has_key(): expects the first argument to be a hash, got #{args[0].inspect} which is of type #{args[0].class}"
     
     Puppet::PreformattedError:
       Evaluation Error: Error while evaluating a Function Call, has_key(): expects the first argument to be a hash, got "" which is of type String (file: /home/aelkiss/Repositories/nebula/spec/fixtures/modules/nebula/manifests/profile/networking/private.pp, line: 22, column: 9) on node qoo.umdl.umich.edu
     # ./spec/fixtures/modules/stdlib/lib/puppet/parser/functions/has_key.rb:22:in `block in <module:Functions>'
     # ./spec/fixtures/modules/nebula/manifests/profile/networking/private.pp:22:in `stack'
     # :2:in `block in call_function'
     # :in `stack'
     # ./spec/classes/profile/networking/private_spec.rb:39:in `block (5 levels) in <top (required)>'
     # ------------------
     # --- Caused by: ---
     # Puppet::ParseError:
     #   has_key(): expects the first argument to be a hash, got "" which is of type String
     #   ./spec/fixtures/modules/stdlib/lib/puppet/parser/functions/has_key.rb:22:in `block in <module:Functions>'

```

Reply:
Worthy of its own ticket is that rspec-puppet doesn't do a good job of imitating actual facts on its own. I do think we should write some defaults that look legit someday, but in the meantime, you'll have to spoof `$facts['networking']['interfaces']` yourself, even if just as `{}`. Right now, either it's an empty string or puppet defaults to reading `nil` as an empty string.

Reply:
😿

Reply:
Used the built-in `in` operator.